Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -77,6 +77,50 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
         })
     }
 
+    override var keyCommands: [UIKeyCommand]? {
+        return [
+            UIKeyCommand(input: UIKeyInputLeftArrow, modifierFlags: .init(rawValue: 0), action: #selector(self.handleKeyCommand(sender:))),
+            UIKeyCommand(input: UIKeyInputRightArrow, modifierFlags: .init(rawValue: 0), action: #selector(self.handleKeyCommand(sender:)))
+        ]
+    }
+    
+    func handleKeyCommand(sender: UIKeyCommand) {
+        switch sender.input {
+        case UIKeyInputLeftArrow:
+            if completionActive {
+                applyCompletion()
+                
+                // Set the current position to the beginning of the text.
+                selectedTextRange = textRange(from: beginningOfDocument, to: beginningOfDocument)
+            } else if selectedTextRange != nil {
+                if selectedTextRange?.start == beginningOfDocument {
+                    return
+                }
+                
+                let cursorPosition = position(from: (selectedTextRange?.start)!, offset: -1)!
```

Comment:
I'd be careful with explicitly unwrapping this. It might be the case that setting the offset to -1 when we're already at the beginning returns nil. Wrap this in a guard before continuing on to using the position, otherwise just return.